hw2/03-count.js

```diff
@@ -1,3 +1,26 @@
-// Add your code here
+const input = document.getElementById("userip");
+const searchText = document.getElementById("search-text");
 
-input.addEventListener('keydown', handleKeyDown);
+input.addEventListener("keydown", handleKeyDown);
+
+function handleKeyDown(event) {
+	if (event.key === "Enter") {
+		const userInput = input.value.trim();
+
+		if (userInput) {
+			// used innerText instead of textContent to just get the text content of an element
+			//as it is visually rendered on the web page
+			const text = searchText.innerText;
+			// regex constructor consisting user text and global and case-insensitive flag
+			const regex = new RegExp(userInput, "gi");
```

Comment:
you need to include the space before and after the word in the regex, otherwise you will highlight parts of words. If you try searching 'the' you will highlight 'the' in the middle of o[the]r.

Reply:
Added space in regex. Now the string gets identified correctly